src/python-common/ceph/deployment/service_spec.py

```diff
@@ -1199,24 +1196,34 @@ def __init__(self,
         self.auth_protocol = auth_protocol
         self.privacy_protocol = privacy_protocol
 
+    @classmethod
+    def _from_json_impl(cls, json_spec: dict) -> 'SNMPGatewaySpec':
+
+        cpy = json_spec.copy()
+        types = [
+            ('snmp_version', SNMPGatewaySpec.SNMPVersion),
+            ('auth_protocol', SNMPGatewaySpec.SNMPAuthType),
+            ('privacy_protocol', SNMPGatewaySpec.SNMPPrivacyType),
```

Comment:
@adk3798 we should move this table over to the base class anc combine it with 

https://github.com/ceph/ceph/pull/44148/commits/9742e950c8ded9169ad3d07aa396e8607b8aa44b#diff-91ff13f38df816b588aedff4193f02803771e1702b8dabd8f67b3e36d939e570R767-R774

IMO there is no need that we implement this  `_from_json_impl` for every subclass.

Reply:
yeah, we don't have `_from_json_impl` for any of the other services spec classes and there's nothing bout this service I see that should make use have to change that. +1 to combing it with KNOWN_PROPERTIES